applications/dapp_wallet_integration_native_mobile_libraries.md

```diff
@@ -0,0 +1,308 @@
+
+1. Allow Tesseract to cover more use cases: add desktop dApps connectivity, ability to work with dApps running in browsers, etc.
+2. Provide the dApp and Wallet developers with a way to integrate Tesseract as easily as it can get.
+
+Even though covering more use cases ASAP sounds tempting, we believe that the UX improvements Tesseract’s current version brings to the Substrate community already provide enough value by itself to focus on and implement an easy way for the dApps and wallets to integrate Tesseract - a set of native libraries (Swift and Kotlin), that allows using Tesseract no harder than any native library the developers are used to.
+
+### Why are the Tesseract native libraries necessary?
+
+iOS and Android provide their developers with native programming languages, which are the standard for the platforms: Swift and Kotlin, respectively.
+
+In contrast, Tesseract is built in Rust for its reliability, robustness, and cross-platform nature. While providing a lot of pros, Rust is quite hard and cumbersome to integrate into a mobile app (dApp or Wallet) and requires a lot of knowledge and effort, which is a significant barrier to Tesseract integration.
+
+To counteract this barrier, we propose to build Swift and Kotlin native libraries that wrap all the complexity of Rust integration under the hood and provide the dApp and Wallet developers with easy and familiar native APIs that require no more effort than any other mobile library.
+
+This way, we reduce the integration barrier to the minimum, allowing Tesseract to participate in Polkadot/Substrate community without bearing any additional complexity for the developer.
+
+### What about cross-platform mobile apps (React Native, Flutter, etc.)?
+
+Cross-platform mobile frameworks, such as React Native and Flutter, are popular among mobile developers. Providing libraries for these frameworks is definitely planned.
+
+However, building a library for such a cross-platform framework is done via Swift/Kotlin bridges. This means that to build the APIs for the cross-platform frameworks, we first need to implement Swift/Kotlin Tesseract libraries. It’s an unavoidable prerequisite.
+
+We are approaching the infrastructure around Tesseract in stages. Even though we would love to have React Native and Flutter asap, to avoid the grant bloating, we narrowed the scope down to Swift/Kotlin, which provides significant value by itself, allowing non-cross-platform developers to benefit from Tesseract while being a prerequisite step for cross-platform libraries development anyhow. React Native and Flutter are the next step.
+
+### Native Libraries APIs
+
+In total, there are four sets of APIs of Tesseract libraries to work on:
+
+1. Android dApp side
+2. Android Wallet side
+3. iOS dApp side
+4. iOS Wallet side
+
+Each of the sides requires several rust libraries to be wrapped:
+
+1. Tesseract itself
+2. IPC transport
+3. Test protocol
+4. Substrate protocol
```

Comment:
We don't aim to wrap the subxt library. For the integration with Substrate, what we really need is a substrate-specific communication protocol. As for the integration with subxt, here is an example of how we did it in our demo apps: https://github.com/tesseract-one/polkachat.rs/blob/main/common/src/substrate/signer.rs

Also, we are providing a Substrate Swift library (not even a wrapper) for iOS developers so that they can use Substrate really easily together with Tesseract: https://github.com/tesseract-one/Substrate.swift/